tests/integrationv2/utils.py

```diff
@@ -105,10 +111,6 @@ def invalid_test_parameters(*args, **kwargs):
     # If we are using a cipher that depends on a specific certificate algorithm
     # deselect the test if the wrong certificate is used.
     if certificate is not None:
-        if protocol is not None:
-            for provider_ in providers:
-                if provider_.supports_protocol(protocol, with_cert=certificate) is False:
-                    return True
```

Comment:
`supports_protocol()` is already called on each provider above, on line 85. So now that the certificate check is separate, this duplicate call to `supports_protocol()` is no longer needed.